core/trino-spi/src/main/java/io/trino/spi/StandardErrorCode.java

```diff
@@ -144,6 +144,7 @@
     INVALID_COPARTITIONING(120, USER_ERROR),
     INVALID_TABLE_FUNCTION_INVOCATION(121, USER_ERROR),
     DUPLICATE_RANGE_VARIABLE(122, USER_ERROR),
+    INVALID_REQUIRED_COLUMNS(123, USER_ERROR),
```

Comment:
The way this is being used above doesn't indicate a user error -- it's an implementation error from whoever wrote the function. Also, we already have a `FUNCTION_IMPLEMENTATION_ERROR` entry, so use that one instead.